### PR TITLE
kubeadm: update the 1.14 HA guide

### PR DESCRIPTION
**Pull request by neolit123**
previous PR:
https://github.com/kubernetes/website/pull/12923

applied all the requested changes from there.

xref: https://github.com/kubernetes/kubeadm/issues/1422

@kubernetes/sig-cluster-lifecycle-pr-reviews 
/assign @jimangel @fabriziopandini 
/priority critical-urgent

### CODE REVIEW

File: content/en/docs/setup/independent/high-availability.md [hunk 1]
```diff
@@ -160,141 +109,126 @@ SSH is required if you want to control all nodes from a single machine.
...
         controlPlaneEndpoint: "LOAD_BALANCER_DNS:LOAD_BALANCER_PORT"
 
     - `kubernetesVersion` should be set to the Kubernetes version to use. This
   example uses `stable`.
     - `controlPlaneEndpoint` should match the address or DNS and port of the load balancer.
     - It's recommended that the versions of kubeadm, kubelet, kubectl and Kubernetes match.
 
-1.  Make sure that the node is in a clean state:
+    {{< note >}}
+    Some CNI network plugins like Calico require a CIDR such as `192.168.0.0/16` and
```
Comment (fabriziopandini) [inline content/en/docs/setup/independent/high-availability.md]: In preview this is rendering as code 🤔

Comment (fabriziopandini) [reply content/en/docs/setup/independent/high-availability.md]: Might be notes should not be indented?

File: content/en/docs/setup/independent/high-availability.md [hunk 2]
```diff
@@ -160,141 +109,126 @@ SSH is required if you want to control all nodes from a single machine.
...
 
-    ```sh
-    kubectl apply -f "https://cloud.weave.works/k8s/net?k8s-version=$(kubectl version | base64 | tr -d '\n')"
-    ```
+      ```sh
+      sudo kubeadm init phase upload-certs --experimental-upload-certs
+      ```
 
-1.  Type the following and watch the pods of the components get started:
+    {{< caution >}}
```
Comment (fabriziopandini) [inline content/en/docs/setup/independent/high-availability.md]: Same here

File: content/en/docs/setup/independent/high-availability.md [hunk 3]
```diff
@@ -306,49 +240,130 @@ the creation of additional nodes could fail due to a lack of required SANs.
                 certFile: /etc/kubernetes/pki/apiserver-etcd-client.crt
                 keyFile: /etc/kubernetes/pki/apiserver-etcd-client.key
 
-    - The difference between stacked etcd and external etcd here is that we are using the `external` field for `etcd` in the kubeadm config. In the case of the stacked etcd topology this is managed automatically.
+      {{< note >}}
```
Comment (fabriziopandini) [inline content/en/docs/setup/independent/high-availability.md]: Same here

File: content/en/docs/setup/independent/high-availability.md [hunk 4]
```diff
@@ -306,49 +240,131 @@ the creation of additional nodes could fail due to a lack of required SANs.
...
+you are going to have to manually copy the certificates from the primary control plane node to the
+joining control plane nodes.
+
+There are many ways to do this. In the following example we are using `ssh` and `scp`:
+
+SSH is required if you want to control all nodes from a single machine.
 
-Each worker node can now be joined to the cluster with the command returned from any of the
-`kubeadm init` commands. The flag `--experimental-control-plane` should not be added to worker nodes.
+1.  Enable ssh-agent on your main device that has access to all other nodes in
```
Comment (sftim) [inline content/en/docs/setup/independent/high-availability.md]: Would it be simpler to document using `scp -3` and skip using an SSH agent?

(`scp -3` runs all transfers via the local client).

Comment (neolit123) [reply content/en/docs/setup/independent/high-availability.md]: in the case of `scp -3` routing we do avoid the need for ssh-agent.
but the idea of enabling ssh-agent is to also allow to ssh to the nodes and execute commands on them.

File: content/en/docs/setup/independent/high-availability.md [hunk 5]
```diff
@@ -19,15 +19,12 @@ control plane nodes and etcd members are separated.
...
-in future versions. You might encounter issues with upgrading your clusters, for example.
+You should also be aware that setting up HA clusters with kubeadm is still experimental and will be further
+simplified in future versions. You might encounter issues with upgrading your clusters, for example.
 We encourage you to try either approach, and provide us with feedback in the kubeadm
 [issue tracker](https://github.com/kubernetes/kubeadm/issues/new).
 
-Note that the alpha feature gate `HighAvailability` is deprecated in v1.12 and removed in v1.13.
-
-See also [The HA upgrade documentation](/docs/tasks/administer-cluster/kubeadm/kubeadm-upgrade-ha-1-13).
+See also [The HA upgrade documentation](/docs/tasks/administer-cluster/kubeadm/kubeadm-upgrade-ha-1-14).
```
Comment (detiber) [inline content/en/docs/setup/independent/high-availability.md]: Isn't the HA upgrade documentation being consolidated with the standard upgrade documentation?

File: content/en/docs/setup/independent/high-availability.md [hunk 6]
```diff
@@ -160,141 +109,126 @@ SSH is required if you want to control all nodes from a single machine.
         apiVersion: kubeadm.k8s.io/v1beta1
         kind: ClusterConfiguration
         kubernetesVersion: stable
-        apiServer:
-          certSANs:
-          - "LOAD_BALANCER_DNS"
```
Comment (detiber) [inline content/en/docs/setup/independent/high-availability.md]: Is this SAN generated automatically now? I know in the past it wasn't.

File: content/en/docs/setup/independent/high-availability.md [hunk 7]
```diff
@@ -160,141 +109,126 @@ SSH is required if you want to control all nodes from a single machine.
...
-    kubectl get pod -n kube-system -w
-    ```
-
-    - It's recommended that you join new control plane nodes only after the first node has finished initializing.
-
-1.  Copy the certificate files from the first control plane node to the rest:
+    - Copy this output to a text file. You will need it later to join control plane and worker nodes to the cluster.
+    - When `--experimental-upload-certs` is used with `kubeadm init`, the certificates of the primary control plane
+    are encrypted and uploaded in the `kubeadm-certs` Secret.
+    - Please note that the `kubeadm-certs` Secret and decryption key expire after two hours. To re-upload the certificates
```
Comment (detiber) [inline content/en/docs/setup/independent/high-availability.md]: This should probably be called out in a note tag to make it more obvious.

File: content/en/docs/setup/independent/high-availability.md [hunk 8]
```diff
@@ -160,141 +109,126 @@ SSH is required if you want to control all nodes from a single machine.
...
+    and generate a new decryption key, use the following command on a control plane that is already joined the cluster:
```
Comment (detiber) [inline content/en/docs/setup/independent/high-availability.md]:
```suggestion
    and generate a new decryption key, use the following command on a control plane node that is already joined to the cluster:
```

File: content/en/docs/setup/independent/high-availability.md [hunk 9]
```diff
@@ -160,141 +109,126 @@ SSH is required if you want to control all nodes from a single machine.
...
-the creation of additional nodes could fail due to a lack of required SANs.
+As stated in the command output, please note that the certificate-key gives access to cluster sensitive data, keep it secret!
 {{< /caution >}}
 
-### Steps for the rest of the control plane nodes
+1.  Apply the CNI plugin of your choice:
 
-1.  Move the files created by the previous step where `scp` was used:
+    [Follow these instructions](/docs/setup/independent/create-cluster-kubeadm/#pod-network) to install
+    the pod network. Make sure this corresponds to whichever pod CIDR you provided in the kubeadm configuration file.
```
Comment (detiber) [inline content/en/docs/setup/independent/high-availability.md]:
```suggestion
    the CNI provider. Make sure the configuration corresponds to the pod CIDR specified in the kubeadm configuration file if applicable.
```

File: content/en/docs/setup/independent/high-availability.md (hunk 6 above)
Comment (fabriziopandini) [reply content/en/docs/setup/independent/high-availability.md]: yes, the code is [here](https://github.com/kubernetes/kubernetes/blob/df2094b3d728bd58c5f23b41add109e32fc7c301/cmd/kubeadm/app/util/pkiutil/pki_helpers.go#L386)

**Comment by neolit123** [comment]
added latest changes requested by @detiber

**Pull request merged**